src/expr/src/scalar/func.rs

```diff
@@ -1483,6 +1483,37 @@ fn rescale_apd<'a>(a: Datum<'a>, scale: u8) -> Result<Datum<'a>, EvalError> {
     Ok(Datum::APD(d))
 }
 
+fn pg_column_size<'a>(a: Datum<'a>) -> Result<Datum<'a>, EvalError> {
+    let sz = repr::datum_size(&a);
+    let sz = match i32::try_from(sz) {
+        Ok(sz) => sz,
+        Err(_) => return Err(EvalError::Int32OutOfRange),
+    };
+    Ok(Datum::Int32(sz))
+}
+
+fn mz_row_size<'a>(a: Datum<'a>) -> Result<Datum<'a>, EvalError> {
+    let sz = match a {
+        Datum::List(a) => {
+            // For record types (which can be full table rows), include the row header size
+            // since in that case users are trying to figure out how much memory the entire
+            // row is taking.
+            //
+            // Using datums_size instead of a.data().len() here is safer because it will
+            // return the size of the datams if they were packed into a Row. Although
+            // a.data().len() happens to give the correct answer (and is faster), data()
+            // is documented as for debugging only.
+            std::mem::size_of::<Row>() + repr::datums_size(a.iter())
```

Comment:
Minor, but at the moment this is sometimes an overestimate, in that things that fit in the inline allocation for `Row` will just use `std::mem::size_of::<Row>()` and only add once they spill out of that. This is a bit dependent on us maintaining that implementation, though. I could see just not adding the `Row` size in, as one very sensible thing that doesn't need to be kept in sync with implementation details.

Reply:
Let's just code this up. I think it's `max(size_of::<Row>()`, repr::datums_size(a.iter())`?

Reply:
I've attempted to implement this so it should be correct in all cases? What'd I miss this time?

Reply:
I was curious about this and took a look at the `SmallVec` implementation. I'm also not sure whether we overestimate, but I think we might underestimate, because we don't consider the actual size of the allocation that happens to hold `sz` bytes (e.g. next power of two)... which might be an argument to just use `sz` and call it a day. I might also be totally wrong about this 😄

Reply:
I agree with this read of the smallvec code, and so we could underestimate. We already know we underestimate because there are multiple copies of Rows and we wanted this function as a lower bound so users can multiple number of rows * mz_row_size and get something. Something someone could improve in the future.

Reply:
I think this underestimation is fine, btw. There are other representations (e.g. the `columnation` stuff) that don't have the same backing allocation, and keeping it as straightforward as possible seems great.

Reply:
> We already know we underestimate because there are multiple copies of Rows and we wanted this function as a lower bound so users can multiple number of rows * mz_row_size and get something.

👍🏼 

> I think this underestimation is fine, btw. 

Same here! Didn't mean to imply that we should try to be "exact" here 😄